core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -393,11 +425,8 @@ public void transportTerminated() {
         // TODO(notcarl): replace this with something more meaningful
         transportShutdown(Status.UNKNOWN.withDescription("transport shutdown for unknown reason"));
         transports.remove(transport);
-        if (shutdown && transports.isEmpty()) {
-          if (terminated) {
-            log.warning("transportTerminated called after already terminated");
-          }
-          terminated = true;
+        if (isTerminated()) {
+          log.warning("transportTerminated called after already terminated");
```

Comment:
This warning is no longer appropriate (at least in its current location). It was detecting if the transport was terminated _before_ the callback was called. The equivalent would be to check `isTerminated()` nearer the beggining of the method.